common/djangoapps/util/tests/test_db.py

```diff
@@ -137,6 +135,45 @@ def do_nothing():
             with atomic():
                 commit_on_success(read_committed=True)(do_nothing)()
 
+    def test_named_outer_atomic_nesting(self):
+        """
+        Test that a named outer_atomic raises an error only if nested in
+        enable_outer_atomic and inside another atomic.
+        """
+        if connection.vendor != 'mysql':
+            raise unittest.SkipTest('Only works on MySQL.')
+
+        outer_atomic(name='abc')(do_nothing)()
+
+        with atomic():
```

Comment:
Yes, it will still throw an error because the `outer_atomic` is not named. Line 110 in this file verifies this.